builder/slide_and_fold_storage.cc

```diff
@@ -107,9 +108,31 @@ class slide_and_fold : public node_mutator {
     int max_workers;
     std::unique_ptr<symbol_map<box_expr>> buffer_bounds;
 
-    loop_info(symbol_id sym, expr orig_min, interval_expr bounds, expr step, int max_workers)
+    var semaphores;
+    var worker_count;
+
+    // How many stages we've added synchronization for.
+    int sync_stages = 0;
+    // If we need synchronization around this stage, which stage it is.
+    std::optional<int> stage;
+
+    bool add_synchronization() {
+      if (sync_stages + 1 >= max_workers) {
```

Comment:
I don't understand why the number of stages is limited by number of workers? Aren't the entire loop body processed by the same worker?